rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -3113,6 +3116,224 @@ private ABICallSpec getAddMultiKeysVote(FederationMember federationMember) {
+            assertLogCommitFederation(oldFederation, newFederation);
+
+            assertCallWasSuccessful(commitFederationCall);
+        }
+
+        @Test
+        void commitFederation_postRSKIP186_preRSKIP419_shouldPerformLegacyCommitFederationActions() {
+            // arrange
+            activations = ActivationConfigsForTest.arrowhead631().forBlock(0L);
+            bridgeEventLogger = new BridgeEventLoggerImpl(bridgeConstants, activations, logs, signatureCache);
+            federationSupport = new FederationSupportImpl(federationMainnetConstants, federationStorageProvider, rskExecutionBlock, activations);
+            List<UTXO> utxosToMove = new ArrayList<>(federationStorageProvider.getNewFederationBtcUTXOs(federationMainnetConstants.getBtcParams(), activations));
+
+            // act
+            FederationChangeResponseCode commitFederationCall = federationSupport.commitFederation(false, pendingFederation.getHash(), bridgeEventLogger);
+
+
+            // assertions
+
+            assertUTXOsWereMoved(utxosToMove);
+
+            assertNewAndOldFederationsWereSet();
+
+            assertPendingFederationVotingWasCleaned();
+
+            assertFederationChangeInfoWasSet();
+
+            Federation oldFederation = federationStorageProvider.getOldFederation(federationMainnetConstants, activations);
+            Federation newFederation = federationStorageProvider.getNewFederation(federationMainnetConstants, activations);
+            assertLogCommitFederation(oldFederation, newFederation);
+
+            assertCallWasSuccessful(commitFederationCall);
+        }
+
+        @Test
+        void commitFederation_postRSKIP419_shouldCallCommitPendingFederation() {
+            // arrange
+            activations = ActivationConfigsForTest.all().forBlock(0L);
+            bridgeEventLogger = new BridgeEventLoggerImpl(bridgeConstants, activations, logs, signatureCache);
+            federationSupport = new FederationSupportImpl(federationMainnetConstants, federationStorageProvider, rskExecutionBlock, activations);
+            Federation activeFederation = federationSupport.getActiveFederation();
+            Federation federationBuiltFromPendingFederation =
+                pendingFederation.buildFederation(Instant.ofEpochMilli(rskExecutionBlockTimestamp), rskExecutionBlockNumber, federationMainnetConstants, activations);
+
+            // act
+            FederationChangeResponseCode commitFederationCall = federationSupport.commitFederation(false, pendingFederation.getHash(), bridgeEventLogger);
+
+
+            // assertions
+
+            // assert proposed federation was set correctly
+            Optional<Federation> proposedFederation =
+                federationStorageProvider.getProposedFederation(federationMainnetConstants, activations);
+            assertTrue(proposedFederation.isPresent());
+            assertEquals(federationBuiltFromPendingFederation, proposedFederation.get());
+
```

Comment:
Assert new and old federations didn't change?

Reply:
good one, thanks :)

Reply:
done